gdbplus/gdb-12.1/gdb/decode.c

```diff
@@ -0,0 +1,1841 @@
+/*
+ * decode.c
+ *
+ *  Created on: Aug 22, 2014
```

Comment:
It is hard to spot there is any different with the one in the `gdbplus/gdb9-2/gdb/decode.c`.
Do we have some better way to migrate this? Could we also create a interface for the gdb like `GDBInterface` and the core analyzer only talks to the `GDBInterface` while it needs to call the gdb function? I'm thinking with that maybe in the future we can support `lldb` by adding add a `LLDBInterface` though it depends on if `gdb` and `lldb` can provide similar functions.